src/generate-from-template.js

```diff
@@ -74,13 +74,18 @@ const generateFromTemplate = async ({ templateName }) => {
     }
   ));
 
-  // Install and build the module
+  // Initialize git before installing deps. This allows git hooks to be setup
+  // as part of install
   log.goToStep(4, templateBanner);
-  await installModule(`./${templateValues.projectName}`);
+  await initializeGitRepo(`./${templateValues.projectName}`);
 
-  // Initialize git
+  // Install and build the module
   log.goToStep(5, templateBanner);
-  await initializeGitRepo(`./${templateValues.projectName}`, generatorOptions);
+  await installModule(`./${templateValues.projectName}`);
+
+  // Create the first commit
+  log.goToStep(6, templateBanner);
+  await createInitialCommit(`./${templateValues.projectName}`, generatorOptions);
```

Comment:
What if users don't want the generated content to be committed?

Reply:
we could can add an opt out flag as an additional feature